kingfisher_scrapy/base_spider.py

```diff
@@ -657,3 +660,71 @@ def _build_url(self, params):
         url_params = params.copy()
         url_params.update(self.additional_params)
         return util.replace_parameters(self.base_url, **url_params)
+
+
+class FlattenSpider(SimpleSpider):
+    """
+    This class makes it easy to collect data from CSV or XLSX files. Uses the ``unflatten`` command in Flatten Tool
+    for OCDS to convert a populated spreadsheet to JSON.
+
+    1. Inherit from ``FlattenSpider``
+    1. Set a ``data_type`` class attribute to the data type of the API responses
+    1. Set a ``ocds_version`` class attribute to the relevant schema version, '1.0' or '1.1' (default).
+    1. Write a ``start_requests`` method (and any intermediate callbacks) to send requests
+
+    .. code-block:: python
+
+        import scrapy
+
+        from kingfisher_scrapy.base_spider import FlattenSpider
+
+        class MySpider(FlattenSpider):
+            name = 'my_spider'
+            data_type = 'release_list'
+
+            def start_requests(self):
+                yield scrapy.Request('https://example.com/api/releases.csv', meta={'file_name': 'all.json'})
+    """
+
+    schema_version_choices = {
+        '1.0': 'https://standard.open-contracting.org/schema/1__0__3/release-schema.json',
+        '1.1': 'https://standard.open-contracting.org/schema/1__1__5/release-schema.json',
+    }
+
+    @handle_http_error
+    def parse(self, response):
+        file_name = urlsplit(response.request.url).path.split('/')[-1:][0]
```

Comment:
Why `[-1:][0]` instead of:

```suggestion
        file_name = urlsplit(response.request.url).path.rsplit('/', 1)[-1]
```